scipy/__init__.py

```diff
@@ -85,6 +85,13 @@
 del linalg
 __all__.remove('linalg')
 
+# The existence of the scipy.fft function breaks doc builds
+import sys
+if 'sphinx' in sys.modules:
```

Comment:
This is not the most robust check. A slightly more robust one would be to do `os.environ['_SCIPY_BUILDING_DOC'] = 'true'` in `doc/source/conf.py` (before any `scipy` import) and then here `if os.getenv('_SCIPY_BUILDING_DOC', '') == 'true'` or so.